### PR TITLE
Log-shipping fix: adopt crosswordv2 log settings

### PR DESCRIPTION
**Pull request by bryophyta**
## What does this change?

Copies the logback config that we're using in [crosswordv2](https://github.com/guardian/crosswordv2/commit/ad3cd755a1b04e94936fec2c18f1574b66cfb2d1)

It includes a number of changes to `build.sbt` as well. It might be that not all of these changes are necessary but I don't understand the jackson eviction stuff well enough to know what's needed and what's not.

Initially created because we thought that the broken log shipping was due to our config, though have since realised that there was an upstream issue with logging. Nevertheless, could be a QoL improvement to have nicer logs?

## Accessibility

-   [ ] [Tested with screen reader](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#screen-reader)
-   [ ] [Navigable with keyboard](https://github.com/guardian/accessibility/blob/main/people-and-technology/02-physical.md#keyboard)
-   [ ] [Colour contrast passed](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#contrast)
-   [ ] [The change doesn't use only colour to convey meaning](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#use-of-colour)

### CODE REVIEW

**Comment by twrichards** [comment]
do we know if this ensures log output is JSON? (so it gets parsed in a structured way by ELK)

similarly does it support stack traces (currently they come out as separate lines, which makes debugging v. painful)

**Comment by bryophyta** [comment]
> do we know if this ensures log output is JSON? (so it gets parsed in a structured way by ELK)
> 
> similarly does it support stack traces (currently they come out as separate lines, which makes debugging v. painful)

If I'm honest I don't understand all the wizardry which @andrew-nowak put into this for crosswords, but in my experience it's worked really nicely there. An example record from crosswordv2 with separate `message` and `stackTrace` fields can be seen here: https://logs.gutools.co.uk/s/editorial-tools/app/discover#/doc/67604290-baa6-11e9-bea2-633437abb232/logstash-ed-tools-2024.09.30?id=T0ajQpIBvaA8WQZa0ffv

File: newswires/conf/logback.xml [hunk 1]
```diff
@@ -1,50 +1,50 @@
...
-         which otherwise risk being sent ANSI escape sequences that they cannot interpret.
-         See https://logback.qos.ch/manual/layouts.html#coloring
-    -->
-    <!-- <withJansi>true</withJansi> -->
-    <encoder class="PatternLayoutEncoder">
-      <charset>UTF-8</charset>
-      <pattern>%d{yyyy-MM-dd HH:mm:ss} %highlight(%-5level) %cyan(%logger{36}) %magenta(%X{pekkoSource}) %msg%n</pattern>
-    </encoder>
+  <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
+    <if condition='property("STAGE").contains("DEV")'>
```
Comment (andrew-nowak) [inline newswires/conf/logback.xml:15]: add the .sbtopts file to make this work https://github.com/guardian/crosswordv2/blob/main/.sbtopts (this condition turns _off_ json encoding for running locally, which is much more human readable)

Comment (bryophyta) [reply newswires/conf/logback.xml:15]: ty! https://github.com/guardian/newswires/pull/31/commits/93a41c8dba21516e4ca5a945a6155371e58d4d07

**Review by andrew-nowak — APPROVED** [review]

**Pull request merged**